protocol-rpc/src/rpc/private-id/rpc_server/metrics.rs

```diff
@@ -0,0 +1,98 @@
+//  Copyright (c) Facebook, Inc. and its affiliates.
+//  SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Serialize};
+use std::sync::{Arc, RwLock};
+use serde_json;
+use std::fs::File;
+
+#[derive(Serialize, Deserialize)]
+struct RawMetrics {
+    protocol_name: String,
+    partner_input_size: Option<usize>,
+    publisher_input_size: Option<usize>,
+    union_file_size: Option<usize>,
+}
+
+impl RawMetrics {
+    fn new(
+        protocol_name: String,
+        partner_input_size: Option<usize>,
+        publisher_input_size: Option<usize>,
+        union_file_size: Option<usize>,
+    ) -> RawMetrics {
+        RawMetrics {
+            protocol_name,
+            partner_input_size,
+            publisher_input_size,
+            union_file_size,
+        }
+    }
+
+    fn save_metrics(&self, path: &str) ->  Result<(), serde_json::Error> {
+        let f = &File::create(path).unwrap();
+        serde_json::to_writer(f, &self)
+    }
+
+    fn print_metrics(&self) {
+        println!("-----BEGIN METRIC VIEW-----");
+        println!("{}", serde_json::to_string(&self).unwrap());
+        println!("-----END METRIC VIEW-----");
+    }
+}
+
+pub struct Metrics {
+    protocol_name: String,
```

Comment:
Given this is written by a single `async` function and not multiple - you can just have them as `usize` - will make the code simpler :)

Reply:
For this, I imitated implementation of `CompanyPrivateId ` in `company.rs`. What is the difference between `CompanyPrivateId` and this?
If it's not running in multiple, we may don't need `Arc` but don't we still need a `RwLock`? Otherwise, with the current implementation, we need to make `PrivateIdService` mutable on https://github.com/facebookresearch/Private-ID/blob/main/protocol-rpc/src/rpc/private-id/server.rs#L155.

Reply:
you are correct :)